codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/awssdk/shapevisitor/conversionwriters/DafnyToAwsSdkConversionFunctionWriter.java

```diff
@@ -325,14 +336,14 @@ public void writeStringEnumShapeConverter(
                   stringShapeWithEnumTrait,
                   value
                 ),
-                value
+                name
```

Comment:
I'm getting 

```
def com_amazonaws_s3_OptionalObjectAttributes(dafny_input):
    # Convert OptionalObjectAttributes
    if isinstance(dafny_input, OptionalObjectAttributes_RestoreStatus):
        return "RESTORE_STATUS"

    else:
        raise ValueError("No recognized enum value in enum type: " + dafny_input)
```

when it should be

```
def com_amazonaws_s3_OptionalObjectAttributes(dafny_input):
    # Convert OptionalObjectAttributes
    if isinstance(dafny_input, OptionalObjectAttributes_RESTORE__STATUS):
        return "RestoreStatus"

    else:
        raise ValueError("No recognized enum value in enum type: " + dafny_input)
```

Maybe

```suggestion
                value
```

then change line 337 to name?